### PR TITLE
Switch to using the Widget API SDK for Jitsi widgets

### PR DESCRIPTION
**Pull request by turt2live**
**Review with https://github.com/matrix-org/matrix-react-sdk/pull/5171**

Widget API SDK: https://github.com/matrix-org/matrix-widget-api

See react-sdk PR for more information.

### CODE REVIEW

**Comment by turt2live** [comment]
The build failure on types is expected here because that job doesn't resolve the react-sdk dependency.

File: src/vector/jitsi/index.ts [hunk 1]
```diff
@@ -79,22 +101,27 @@ let meetApi: any; // JitsiMeetExternalAPI
...
             // See https://github.com/matrix-org/prosody-mod-auth-matrix-user-verification
             if (jitsiAuth === JITSI_OPENIDTOKEN_JWT_AUTH) {
                 // Request credentials, give callback to continue when received
-                widgetApi.requestOpenIDCredentials(credentialsResponseCallback);
+                openIdToken = await widgetApi.requestOpenIDConnectToken();
+                console.log("Got OpenID Connect token");
+                enableJoinButton();
             } else {
                 enableJoinButton();
             }
```
Comment (t3chguy) [inline src/vector/jitsi/index.ts:115]: the `enableJoinButton()` could probably be outside the if once for readability

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/vector/jitsi/index.ts [hunk 2]
```diff
@@ -146,7 +160,7 @@ function createJWTToken() {
         room: "*",
         context: {
             matrix: {
-                token: widgetApi.openIDCredentials.accessToken,
+                token: openIdToken.access_token,
```
Comment (t3chguy) [inline src/vector/jitsi/index.ts:163]:
```suggestion
                token: openIdToken.access_token, // eslint-disable-line camelcase
```

Comment (turt2live) [reply src/vector/jitsi/index.ts:163]: ah yes, you would think that, wouldn't you. Somehow it doesn't matter - the linter is happy in this area and isn't complaining about `room_id` directly below either. I also can't see a block disabling the rule in the area, so I presume it is magic.

**Pull request merged**